.github/workflows/rust.yml

```diff
@@ -0,0 +1,48 @@
+name: Rust test
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        rust: [stable, nightly, beta]
+
+    steps:
+      - uses: hecrj/setup-rust-action@v1
+        with:
+          rust-version: ${{ matrix.rust }}
+      - uses: actions/checkout@master
+      - name: Build
+        run: cargo build --examples --lib --bins
+      - name: Test
+        run: cargo test --verbose --examples --lib --bins
+  clippy:
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    steps:
+      - uses: actions/checkout@master
+      - name: Install clippy
+        run: rustup component add clippy
+      - name: Run clippy
+        run: cargo clippy --all-targets -- --deny clippy::all
+  fmt:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust: [stable, beta]
+    steps:
+      - uses: hecrj/setup-rust-action@v1
+        with:
+          rust-version: ${{ matrix.rust }}
+      - uses: actions/checkout@master
```

Comment:
does it checkout only the `master` branch? We probably want the CI to work on any branch...

Reply:
That is a reference to version of the actions/checkout, i locked that down to v2 now, and have reconfigured the `on>push>[master]` to `on>push>[*]`

Reply:
ok thanks